src/user/user.service.ts

```diff
@@ -1,298 +1,254 @@
+     * @param email of the user
+     * @returns User
+     */
+    async findOneByEmail(email: string): Promise<User | null> {
+        const user = await this.userSchema.findOne({ email }).lean();
 
-    return result
-  }
+        if (!user) return null;
 
-  /**
-   * Sends the code for a password reset to the mail adress
-   * @param mail - usermail
-   */
-  async requestResetPassword(mail: string) {
-    const user = await this.userSchema.findOne({
-      'email': mail
-    }).lean()
+        return user;
+    }
 
-    if (!user) {
-      throw new NotFoundException()
+    /**
+     * Update the user
+     * @param id ObjectId
+     * @param updateUserDto Dto for updates
+     * @returns updated user (with changed fields)
+     */
+    async updateUser(
+        id: ObjectId,
+        updateUserDto: UpdateUserDto
+    ): Promise<User> {
+        try {
+            const updatedUser: User = await this.userSchema.findByIdAndUpdate(
+                id,
+                {
+                    ...updateUserDto
+                },
+                {
+                    new: true
+                }
+            );
+
+            return updatedUser;
+        } catch (error) {
+            if (error.code === 11000)
+                throw new ConflictException('Username is already taken.');
+            else throw new InternalServerErrorException('Update User failed');
+        }
     }
 
-    const resetCode = crypto.randomBytes(64).toString('hex');
-    const resetObject = new this.resetSchema({
-      userId: user._id,
-      verificationCode: resetCode
-    })
-    const result = await resetObject.save()
+    async remove(id: ObjectId, requestingUser: JwtUserDto): Promise<User> {
+        if (
+            requestingUser.role !== Role.Admin &&
+            id !== requestingUser.userId
+        ) {
+            throw new UnauthorizedException();
+        }
 
-    await this.mailService.sendPasswordReset(user.username, user.email, resetCode)
+        const user = await this.userSchema.findByIdAndDelete(id);
 
-  }
+        if (!user) throw new NotFoundException();
 
-  async findReset(userId: ObjectId): Promise<EmailVerify> {
-    const verifyObject = await this.resetSchema.findOne({
-      'userId': userId
-    }).lean()
-    if (!verifyObject) {
-      throw new NotFoundException()
-    }
-    return verifyObject
-  }
-
-  /**
-   * Overwrites the password with the new one
-   * @param code - resetcode as passed in url
-   * @param password new password
-   * @returns 
-   */
-  async validatePasswordReset(code: string, password: string) {
-    const resetObject = await this.resetSchema.findOneAndDelete({
-      'verificationCode': code
-    })
-
-    if (!resetObject) {
-      throw new NotFoundException()
+        return user;
     }
 
-    if (Date.now() - resetObject._id.getTimestamp() > +process.env.RESET_TTL) {
-      throw new BadRequestException('Token expired.')
+    async validateVerifyCode(userId: ObjectId): Promise<boolean> {
```

Comment:
Many duplicate code to `isValidJWT` maybe we can put them together. On both we have to check only for a Valid User.

Reply:
hard to scale and would probably mean that at least one method has to return a user rather than a boolean...so I would not do this__